xcode/Subconscious/Shared/Components/Settings/SettingsView.swift

```diff
@@ -8,10 +8,39 @@
 import SwiftUI
 import ObservableStore
 
+struct GatewaySyncLabel: View {
+    var status: GatewaySyncStatus
+    @State var spin = false
+    
+    var body: some View {
+        HStack {
+            switch (status) {
+            case .initial:
+                Image(systemName: "arrow.triangle.2.circlepath")
+            case .inProgress:
+                Image(systemName: "arrow.triangle.2.circlepath")
+                    .rotationEffect(.degrees(spin ? 360 : 0))
+                    .animation(Animation.linear
+                        .repeatForever(autoreverses: false)
+                        .speed(0.4), value: spin)
+                    .onAppear() {
+                        self.spin = true
+                    }
+            case .success:
+                Image(systemName: "checkmark.circle")
+            case .failure:
+                Image(systemName: "exclamationmark.arrow.triangle.2.circlepath")
+            }
+            
+            Text("Sync with Gateway")
```

Comment:
Perhaps the label should also change? Might be a bit visually jarring

Reply:
I think the iOS way to do this is usually to have the spinner on the right-hand side of the table cell (secondary icon color).

Reply:
Updated the appearance, it uses secondary color by default but uses the accent color when actively animating.

<img width="369" alt="image" src="https://user-images.githubusercontent.com/5009316/222604645-4e951123-65a7-4f54-b576-2659da2d87a1.png">

<img width="368" alt="Screenshot 2023-03-03 at 10 53 31 am" src="https://user-images.githubusercontent.com/5009316/222604523-a609ff1b-0fcf-43f8-abb6-afa044528d9b.png">

<img width="368" alt="Screenshot 2023-03-03 at 10 52 52 am" src="https://user-images.githubusercontent.com/5009316/222604528-aa59098f-0e92-49cc-85dc-4b46d343cd98.png">

Reply:
Also tracking the error message now, and rendering a special failure case:

<img width="367" alt="image" src="https://user-images.githubusercontent.com/5009316/222605995-77449616-c2a2-4b58-9d55-809056d7587f.png">